src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -268,8 +268,7 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
     List<TaskInfo> tasks = new ArrayList<>();
     for (String taskType : taskTypes) {
       List<Resource> taskResources = getTaskResources(taskType);
-      String taskName = getNextTaskType(taskType);
-      if (taskName.isEmpty()) return false;
+      String taskName = getNextTaskName(taskType);
```

Comment:
Since getNextTaskName can now throw an exception (instead of returning empty), should we try/catch that exception here, or is it fine to let it propagate up?